ionoscloud/resource_autoscaling_group.go

```diff
@@ -0,0 +1,484 @@
+
+	group, err := autoscalingService.GetAutoscalingGroupDataCreate(d)
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("an error occured getting data from the provided schema: %w", err))
+	}
+	if group.Properties != nil && group.Properties.Name != nil {
+		log.Printf("[DEBUG] autoscaling group data extracted: %+v", *group.Properties.Name)
+	}
+
+	autoscalingGroup, _, err := client.CreateGroup(ctx, *group)
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("error creating autoscaling group: %w", err))
+		return diags
+	}
+
+	d.SetId(*autoscalingGroup.Id)
+	log.Printf("[INFO] autoscaling Group created. Id set to %s", *autoscalingGroup.Id)
+
+	if err := checkAction(ctx, client, d); err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+	if err := autoscalingService.SetAutoscalingGroupData(d, autoscalingGroup.Properties); err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceAutoscalingGroupRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+
+	client := meta.(services.SdkBundle).AutoscalingClient
+
+	group, apiResponse, err := client.GetGroup(ctx, d.Id(), 2)
+	if err != nil {
+		if apiResponse.HttpNotFound() {
+			log.Printf("[INFO] resource %s not found: %+v", d.Id(), err)
+			d.SetId("")
+			return nil
+		} else {
+			diags := diag.FromErr(fmt.Errorf("error while retrieving autoscaling group with id %v, %w", d.Id(), err))
+			return diags
+		}
+	}
+
+	log.Printf("[INFO] successfully retrieved autoscaling group %s: %+v", d.Id(), group)
+
+	if err := autoscalingService.SetAutoscalingGroupData(d, group.Properties); err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] successfully set autoscaling group data %s", d.Id())
+
+	return nil
+}
+
+func resourceAutoscalingGroupUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+
+	client := meta.(services.SdkBundle).AutoscalingClient
+
+	group, err := autoscalingService.GetAutoscalingGroupDataUpdate(d)
+
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	if group.Properties != nil && group.Properties.Name != nil {
+		log.Printf("[DEBUG] autoscaling group data extracted: %s", *group.Properties.Name)
+	}
+	updatedGroup, _, err := client.UpdateGroup(ctx, d.Id(), *group)
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("an error occured while updating autoscaling group %s: %w", d.Id(), err))
+		return diags
+	}
+	log.Printf("[INFO] autoscaling Group updated.")
+
+	if err := checkAction(ctx, client, d); err != nil {
+		return diag.FromErr(err)
+	}
+
+	return diag.FromErr(autoscalingService.SetAutoscalingGroupData(d, updatedGroup.Properties))
+}
+
+func resourceAutoscalingGroupDelete(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
+	client := meta.(services.SdkBundle).AutoscalingClient
+
+	_, err := client.DeleteGroup(ctx, d.Id())
+	if err != nil {
+		diags := diag.FromErr(fmt.Errorf("an error occured while deleting an autoscaling group %s %w", d.Id(), err))
```

Comment:
return `diag.FromErr` directly

Reply:
@cbeti-ionos solved in c9df22e0066ca69a7809de59cd124fe97fec9580